doc/rtd/reference/datasources/wsl.rst

```diff
@@ -0,0 +1,205 @@
+.. _datasource_wsl:
+
+WSL
+***
+
+The Windows Subsystem for Linux (WSL) somewhat resembles a container
+hypervisor, a Windows user may have as many Linux distro instances as they
+wish, either created by the distro-launcher workflow (for the distros delivered
+through MS Store) or by importing a tarball containing a root filesystem. This
+page assumes the reader is familiar with WSL. To learn more about that, please
+visit the `Microsoft documentation <https://learn.microsoft.com/windows/wsl/about>`_.
+
+Requirements
+==============
+
+1. **WSL interoperability must be enabled**. The datasource needs to execute
+   some Windows binaries to compute the possible locations of the user data
+   files.
+
+2. **WSL automount must be enabled**. The datasource needs to access files in
+   the Windows host filesystem.
+
+3. **The init system must be aware of cloud-init**. WSL has opt-in support for
+   systemd, thus for distros that rely on it, such as Ubuntu, cloud-init will
+   run automatically if systemd is enabled via the ``/etc/wsl.conf``. The
+   Ubuntu applications distributed via Microsoft Store enable systemd in the
+   first boot, so no action is required if the user sets up a new instance by
+   using them. Users of other distros may find it surprising that cloud-init
+   doesn't run automatically by default. At the time of this writing, only
+   systemd distros are supported by the WSL datasource, although there is
+   nothing hard coded in the implementation code that requires it, so
+   non-systemd distros may find ways to run cloud-init and make it just work.
+
+Notice that requirements 1 and 2 are met by default, i.e. WSL grants those
+features enabled, being the user allowed to opt-out if so desired.
```

Comment:
I found this line to be a little confusingly worded - I don't know exactly what was intended by "being the user allowed to opt-out" so I don't have an alternative suggestion.

Reply:
Thanks for the change, I think it's clearer now :)